services/QuillLMS/app/queries/user_pack_sequence_item_query.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class UserPackSequenceItemQuery
+  FINISHED = ActivitySession::STATE_FINISHED
+  COMPLETED_KEY = 'completed'
+  PACK_SEQUENCE_ITEM_ID_KEY = 'pack_sequence_item_id'
+
+  def self.call(classroom_id, user_id)
```

Comment:
Curious why a `self.call` here instead of the service object `run` pattern you've been using in most of your code. Do you view `queries` differently?

Reply:
Conceptually, I view service objects as a generic computation where a return type that can be nil and there might be side-effects.  I see queries as a SQL wrapper with a return non-nil return type (usually a ActiveRecord::Relation or PG::Result) that can be chained/filtered/composed.

As I work through the different use cases, I feel like the [query object pattern](https://thoughtbot.com/blog/a-case-for-query-objects-in-rails) will probably help, but I don't yet know what that will look like, so I'm just using the existing `self.call` pattern from other code in our codebase.